### ISSUE TITLE
GameGuru MAX - Stand Alone Bug - Game Stutters Initially

### ISSUE THREAD

**Issue by ArgentArts** [TheGameCreators/GameGuruRepo#2300]
When I first start any stand alone game that I export, the game stutters for as long as I stay in my initial position. If I move the camera, but don't walk, the camera is choppy. It's not until I've taken several steps that suddenly everything is smooth as silk and I can move smoothly, both walking and camera movement. Here's a video:

https://user-images.githubusercontent.com/36735507/155826622-0c94dce7-5542-4453-8012-c28a93c5b25c.mp4

This has happened with every single stand alone I've exported. This DOES NOT happen with any test games I test within the MAX editor. So, if I run the game within MAX by pressing the rocket icon, it's smooth as silk right from the beginning. This only happens with stand alone games and only if I don't move the player. If I stand in the same position (don't walk forward, backward, etc.) then the  camera will will stay choppy forever. Once I take several steps in a direction, it all sorts out and everything is fine.

The example level in the video is a simple one that I am getting 100 fps consistently on. So, it's not my PC. Or it shouldn't be.

**Comment by ArgentArts** [comment]
I had opened another report about stand alone games having tons of extra unneeded/unused files in them here:

https://github.com/TheGameCreators/GameGuruRepo/issues/2301

One discovery is that when I had deleted all/most of the directories with these unused files and ran the stand alone game, the stuttering (shown above) completely goes away.

**Comment by ArgentArts** [comment]
Exported another stand alone game. Initially the folder was 1.13GB in size. Ran the game and got the initial stuttering, like I was stuck (just like in the video above). Went through the folder and removed all the extra stuff, which reduced the folder size down to  541MB. Ran the level and it ran perfectly. For whatever reason, these extra files are causing issues with the stand alone. They shouldn't be there to begin with and they are causing huge issues at the beginning of the game.

**Comment by ArgentArts** [comment]
I've narrowed it down a bit to the particular folder that appears to be causing the stuttering. If you remove/delete the "materials" folder under "audiobank", the initial stuttering disappears. I am assuming this is happening because MAX wants to load and have a material sound ready for when the player walks, but that doesn't appear to be working at the moment? So, it's causing some sort of stuttering because this is failing? Just a guess. But completely removing the folder completely gets rid of the stutter and putting the folder back causes the stutter to return.

**Comment by LeeBamberTGC** [comment]
@ArgentArts Great detective work!  @ZakJudges I have pulled this one back as I know the code responsible for this I believe. The code reserves sound slots but only clones the sound 'when needed', but I think it takes a few seconds to create those resources and thus the stutter. Moving the cloning back to the init step should remove the stutter, will find out soon.

**Comment by ArgentArts** [comment]
About the stuttering - if I don't move forward at all (don't walk), but just pan the camera, the stuttering will continue indefinitely. I can literally pan the camera back and forth for minutes (or until I get tired of it) and the stuttering continues. It will not stop literally until I start to move forward and then it takes a few steps forward before the player is released from the stuttering. So, it's like the player is "locked" into this stuttering phase until they move out of that initial location.

**Comment by LeeBamberTGC** [comment]
@ArgentArts Just did some work on materials and sounds, and when I went to reproduce with a simple level I could not get the stutter. Can you try with the current latest internal build just uploaded, I may have accidentally fixed it.

**Comment by ArgentArts** [comment]
It may be a moment ... working on the docs for Rick and have to wait for a break.

BTW - these stutters are happening when I am creating custom graphics on an empty level (all my own models). So, none of the materials are MAX materials. Does that help?

**Comment by ArgentArts** [comment]
How do I update to the latest internal build?

**Comment by LeeBamberTGC** [comment]
@ArgentArts Are you part of our internal testers team?  If not, I can sign you up so you get access to internal builds when you go to the Check For Updates.

**Comment by ArgentArts** [comment]
I am not part of the internal testing team, but am willing to test. :)

**Comment by LeeBamberTGC** [comment]
@ArgentArts Email me at lee@thegamecreators.com with the email you use to log into your TGC account and I will get you signed up :)

**Comment by ArgentArts** [comment]
email sent. :)

**Comment by ArgentArts** [comment]
Please let me know when this is completed and I will test ASAP. Thanks.

**Comment by plemsoft** [comment]
@ArgentArts Could you post a video where the fps is visible , it would really help , mainly i see this  stuttering when the fps gets below 25, so just trying to figure out if this is the case , thanks :)

**Comment by ArgentArts** [comment]
Is there a way to display the FPS counter in the standalone? Or do I need an external tool for that? Because the stutter only happens with the standalone.

**Comment by ArgentArts** [comment]
@LeeBamberTGC Email sent to other email address. ;)

**Comment by ArgentArts** [comment]
Here's a video that has the frame rate in the upper-left:

[http://www.argentarts.com/uploads/stutter2.mp4](url)

As you can see, my frame rate is in the 20s as long as I don't step forward, but just pan the camera. Once I take a few steps, the frame rate shoots up to the 90s. Actually, I cut the video short, because my frame rate tops off at 100fps in this level.

**Comment by ArgentArts** [comment]
@LeeBamberTGC - Are you getting my emails? I've sent a few to both email addresses.

**Comment by LeeBamberTGC** [comment]
@ArgentArts Not getting them, sorry.  I will use your forum name to find your records on the TGC server, leave it with me :)

**Comment by LeeBamberTGC** [comment]
@ArgentArts I found your display name (with space) but could not see the MAX product in your history.  Perhaps you have two TGC accounts?

**Comment by ArgentArts** [comment]
@LeeBamberTGC This is strange, because we've exchanged emails previously. The email I used to sign up for MAX is: contact@argentarts.com.

I'd also sent an email to you because I wanted to find out if you needed someone to create sci-fi weapon models. I'm a 3D modeler (professionally) and would like to learn about helping in that area, if you have the need.

**Comment by ArgentArts** [comment]
The display name is Argent_Arts on the forum and GameCreators site.

**Comment by LeeBamberTGC** [comment]
@ArgentArts Found you. I have added you the internal group, expect an email sometime today about the next internal build :)  Feel free to email again and I will see if "contact@argentarts.com" is being moved to the socials filter :)  New weapons sound fab!!

**Comment by ArgentArts** [comment]
As of today's internal build, the initial stuttering is still there on a standalone game. And, like previously, if I delete the "materials" folder from the "audiobank" folder, the standalone game then runs buttery smooth from the get-go. So, this is still an issue.

Again, this i

![SFH_11](https://user-images.githubusercontent.com/36735507/157680040-7b7e8398-efdc-4921-9b99-b5e1e1251c8d.png)
s an indoor-only level with all custom materials.

**Comment by ArgentArts** [comment]
@LeeBamberTGC - Just wanted to check to see if you'd received the other emails I'd sent you today. Not trying to bug you (know you're super busy). Just wanted to see if my emails are (finally) getting through. Thanks.

**Comment by LeeBamberTGC** [comment]
@ArgentArts Nothing current in my inbox :)  You might need to keep using your personal email address to send :)

**Comment by ArgentArts** [comment]
@LeeBamberTGC - Sent again via personal email. ;)
